### ISSUE TITLE
Reason field in blocks

### ISSUE THREAD

**Issue by koyuawsmbrtn** [tootsuite/mastodon#13548]
### Pitch

If someone blocked somebody on Mastodon wouldn't it be nice to have an optional field explaining why the person has been blocked? This would save a lot of people some headache and keeps them calm. It kind of freaks me out every time I see someone block me for no reason.

### Motivation

This would give people that have been victimized by so-called "block recommendations" the opportunity to find out what they did wrong and how the issue could be mitigated in the future.

**Comment by SuperSandro2000** [comment]
Also giving a 404 is also a very bad user experience cause most people think the server is broken rather than that they are blocked.

**Comment by ThibG** [comment]
The idea of blocks is to be the least noticeable possible while still, well, blocking the targeted user. Of course it's not perfect, and users may have different expectations (see #11562).

I'm not strongly opposed to that feature, but it would require protocol and API changes, and I think very few people would want to make use of it… and people can still DM the blocked person before blocking them (but it would proactively tell them, eh)

@SuperSandro2000 It doesn't give a 404, it gives a profile with no toots.

**Comment by SuperSandro2000** [comment]
> It doesn't give a 404, it gives a profile with no toots.

Which is the equivalent for mastodon server has hickups and failed to fetch that user.

**Comment by ThibG** [comment]
Not really. The equivalent is not having a profile at all. Anyway the point of blocks is not to notify the blocked person that they are, indeed, blocked (even though, in the absence of stealth block, that information may be available to them).

**Comment by SuperSandro2000** [comment]
> Anyway the point of blocks is not to notify the blocked person that they are, indeed, blocked

If the profile is public I can just copy the URL, open a incognito tab and check what they are doing. If the profile is not public I would still know that they blocked me cause I couldn't view their toots.

If you are introducing stealth blocks that my server would not know and a user blocks me and I interact a lot with them than my server essentially spams their server. If I have a bot that toots a lot and someone stealth blocks them I would notify them about every toot and they would throw it away. If blocks are not stealth admins could still find it out especially if they selfhost an instance.

In my opinion it should just clearly be shown that a user blocked you for transparency reasons and less work for admins and mods with user requests that the server has some hickups and fails to communicate with some other server. I had problems in the past where my server no longer could communicated with some instances which did not have me blocked or toot indexing was to slow or failed for some reason.

Also if you think about the perspective of a blocked person most of them will think the server has some sort of problem rather than that they are blocked. Also if someone calls out to block a specific person and they interact with anyone in the future they might or might not be blocked which can be really frustrating especially if the person did not do or want harm.

To summarize that: Either blocks are public for at least admins or you generate a lot of potential wasted traffic and frustration and may trigger firewalls, etc.

**Comment by Kleidukos** [comment]
Wow that was a bad take.

**Comment by SuperSandro2000** [comment]
Well, it is my current opinion and maybe it changes in the future but for now I don't believe in shadow blocks like this and prefer the twitter way where it tells you that person blocked you.

**Comment by shleeable** [comment]
Being able to block people and write a note why YOU blocked them is a great idea -- I regularly forget why I've blocked people.

Them being able to see it? doubt.
